src/appleseed-max-impl/oslutils.cpp

```diff
@@ -41,16 +41,118 @@
 
 // 3ds Max Headers.
 #include <bitmap.h>
+#include <imtl.h>
+#include <maxapi.h>
 #include <stdmat.h>
 
 namespace asf = foundation;
 namespace asr = renderer;
 
+namespace
+{
+    asr::ParamArray get_uv_params(Texmap* texmap)
+    {
+        asr::ParamArray uv_params;
+
+        if (!is_bitmap_texture(texmap))
+            return uv_params;
+
+        UVGen* uv_gen = texmap->GetTheUVGen();
+        if (!uv_gen || !uv_gen->IsStdUVGen())
+            return uv_params;
+
+        StdUVGen* std_uv = static_cast<StdUVGen*>(uv_gen);
+        auto time = GetCOREInterface()->GetTime();
+
+        // Assume that all the textures are in texture mode texmap->MapSlotType() == MAPSLOT_TEXTURE.
```

Comment:
Even better than comments: assertions. Something like

    DbgAssert(texmap->MapSlotType() == MAPSLOT_TEXTURE);
    DbgAssert(static_cast<StdUVGen*>(uv_gen)->GetCoordMapping() == UVMAP_EXPLICIT);

Reply:
Great advice thanks. And in release build those asserts are just ignored? That's what I wanted to happen.

Reply:
Yes, assertions are completely ignored in optimized builds. Which poses a risk, by the way, for instance this wouldn't be a proper way to check that `write_to_file()` worked, because the entire call would be ignored in optimized builds:

    DbgAssert(write_to_file("file.ext"));

Something to keep in mind :)